jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanSessionDataSerializer.java

```diff
@@ -0,0 +1,69 @@
+package org.eclipse.jetty.session.infinispan;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.infinispan.commons.marshall.Externalizer;
+
+public class InfinispanSessionDataSerializer implements Externalizer<InfinispanSessionData>
```

Comment:
Why is this class necessary? We are using the org.eclipse.jetty.session.infinispan.SessionDataMarshaller now, are you saying this class replaces the SessionDataMarshaller?

Reply:
SessionDataMarshaller is ok in the remote setup, but it is not used in embedded infinispan when storeAsBinary is enabled. In this case the default java serialization rules are used, which fail because the SessionData readObject is invoked and ObjectInputStream is not ClassLoadingObjectInputStream

Reply:
OK. Please see my comment above about the presence of the annotation.